src/lib/modules/contracts_manager/index.js

```diff
@@ -291,9 +291,21 @@ class ContractsManager {
           callback();
         });
       },
-      function compileContracts(callback) {
+      function allContractsCompiled(callback) {
+        const allContractsCompiled = 
+          self.compiledContracts && 
+          self.contractsFiles && 
+          self.contractsFiles.every(contractFile => 
+            Object.values(self.compiledContracts).find(contract => 
+              contract.originalFilename === contractFile.filename
+            )
+          );
+        callback(null, allContractsCompiled);
```

Comment:
There is nothing async in this function, why do you use a callback?

It could be a pure function out of the async branch
What do you think?

Reply:
Agreed, it was as per the request of @PaschalPrecht and @jrainville. Could definitely move it out of the waterfall if it’s offensive.

Reply:
I would not say offensive :) it is working and don't have much impact.
This is up to you really!

Reply:
Oh yea, that was definitely a misunderstanding @emizzle sorry, I probably didn't express myself correctly. Pretty much what @jrainville said. The expression that evaluates whether all contracts have been compiled or not could be put into a function, as it's easier to parse for human brains:

```
const allContractsCompiled = allContractsCompiled()
```